src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.Stream.Tests.cs

```diff
@@ -268,22 +268,25 @@ public void UnseekableStreams_RoundTrip(TestTarFormat testFormat)
 
         [Theory]
         [MemberData(nameof(GetExactRootDirMatchCases))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/88049", TestPlatforms.iOS | TestPlatforms.tvOS)]
         public void ExtractToDirectory_ExactRootDirMatch_RegularFile_And_Directory_Throws(TarEntryFormat format, TarEntryType entryType, string fileName)
         {
             ExtractToDirectory_ExactRootDirMatch_RegularFile_And_Directory_Throws_Internal(format, entryType, fileName, inverted: false);
             ExtractToDirectory_ExactRootDirMatch_RegularFile_And_Directory_Throws_Internal(format, entryType, fileName, inverted: true);
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/88049", TestPlatforms.iOS | TestPlatforms.tvOS)]
         public void ExtractToDirectory_ExactRootDirMatch_Directory_Relative_Throws()
         {
             string entryFolderName = "folder";
             string destinationFolderName = "folderSibling";
 
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+            // the folder used to store files needs to have a shorter path on Apple mobile platforms,
+            // because the TempDirectory gets created in folder with a path longer than 100 bytes
+            using TempDirectory root = new TempDirectory("/Users/helix-runner/Library/Developer/CoreSimulator/Devices/tempDir");
```

Comment:
This should work on simulator and maybe catalyst, but should fail on device. As far as I know, there's no way to shrink the app bundle path (where your app gets a tmp directory).

Reply:
According to the `TempDirectory` class, it is possible to specify the full path:
```csharp
public TempDirectory(string path)
{
    Path = path;
    Directory.CreateDirectory(path);
}
```
One concern regarding this is permissions and possibility to create a particular directory.

Reply:
It seems that the `#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS` condition in the library test is evaluated as `FALSE` on Apple mobile targets. Without the macro condition and with shorter temporary directory `using TempDirectory root = new TempDirectory("/Users/helix-runner/tmp");` the tests seems to work on Apple mobile targets.